src/helpers/__tests__/helpers.test.ts

```diff
@@ -48,14 +48,16 @@ describe('getAuthType', () => {
       become_user: undefined,
       become_password: undefined,
       sources: [],
+      auth_type: undefined,
       ...partialCredential
     });
 
   it(`should return a credential type`, () => {
     expect([
-      generateAuthType({ username: 'testUser', password: 'testPassword' }),
-      generateAuthType({ auth_token: 'mockToken' }),
-      generateAuthType({ ssh_key: 'mockSSH' })
+      generateAuthType({ auth_type: 'password' }),
+      generateAuthType({ auth_type: 'auth_token' }),
+      generateAuthType({ auth_type: 'ssh_key' }),
+      generateAuthType({ auth_type: 'ssh_keyfile' })
```

Comment:
optional:
since we've refined everything down to `auth_type` you've opened the ability to loop through the enum values instead of declaring them individually... 

pseudo code
```
Object.values(helpers.authType).map(authType => generateAuthType({ auth_type: authType}))
```